pandas/core/internals/blocks.py

```diff
@@ -2145,6 +2128,32 @@ def get_values(self, dtype=None):
                                  self._box_func).reshape(self.values.shape)
         return self.values
 
+    def _try_coerce_args(self, values, other):
+        self._validate_tz(other)
+
+        if is_null_datelike_scalar(other):
+            other = self.fill_value
+        elif isinstance(other, (Timedelta, tslibs.Timestamp)):
+            other = other.asm8
+        elif isinstance(other, timedelta):
+            other = np.timedelta64(other)
+        elif isinstance(other, date):
+            other = np.datetime64(other)
+        elif isinstance(other, (np.timedelta64, np.datetime64, np.ndarray)):
+            pass
+        elif (hasattr(other, 'dtype')
+              and (is_timedelta64_dtype(other) or is_datetime64_dtype(other))):
+            other = getattr(other, 'values', other)
+        else:
+            # coercion issues
+            # let higher levels handle
+            raise TypeError
+
+        return values, other
```

Comment:
Seems `values` can be removed too. The other 2 `_try_coerce_args` overrides don't touch `values`, `_try_coerce_args` might rename to `_try_coerce_other`.